src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs

```diff
@@ -68,13 +68,21 @@ protected TxDecoder(bool lazyHash = true)
             {
                 if (!rlpStream.IsSequenceNext())
                 {
-                    (int PrefixLength, int ContentLength) prefixAndContentLength =
-                        rlpStream.ReadPrefixAndContentLength();
-                    transactionSequence = rlpStream.Peek(prefixAndContentLength.ContentLength);
+                    (int _, int contentLength) = rlpStream.ReadPrefixAndContentLength();
+                    transactionSequence = rlpStream.Peek(contentLength);
                     transaction.Type = (TxType)rlpStream.ReadByte();
                 }
             }
 
+            int networkWrapperCheck = 0;
+            if ((rlpBehaviors & RlpBehaviors.InNetworkForm) == RlpBehaviors.InNetworkForm && transaction.HasNetworkForm)
```

Comment:
transaction.IsBlobType would simplify it

Reply:
Blob tx can be passed to the decoder in any form, we cannot assume just by tx type